src/hash.ts

```diff
@@ -1,18 +1,43 @@
 import { decoders, encoders } from '@47ng/codec'
 import crypto from 'node:crypto'
-import { HashFieldConfiguration } from 'types'
+import { HashFieldConfiguration, HashFieldSanitizeOptions } from './types'
 
 export function hashString(
   input: string,
   config: Omit<HashFieldConfiguration, 'sourceField'>
 ) {
   const decode = decoders[config.inputEncoding]
   const encode = encoders[config.outputEncoding]
-  const data = decode(input)
+  const sanitized = sanitizeHashString(input, config.sanitize)
+
+  const data = decode(sanitized)
   const hash = crypto.createHash(config.algorithm)
   hash.update(data)
   if (config.salt) {
     hash.update(decode(config.salt))
   }
   return encode(hash.digest())
 }
+
+export function sanitizeHashString(
+  input: string,
+  options: HashFieldSanitizeOptions[] = []
+) {
+  let output = input
+  if (options.includes(HashFieldSanitizeOptions.lowercase)) {
+    output = output.toLowerCase()
```

Comment:
I'm not sure. But my first thought would be no. If we use `toLocaleLowerCase` it will start to use the locale of your api/backend system, where in this case you actually want to flatten your hash input always in the same way.
I'm not completely sure, but I cannot come up with an example wherein `toLocaleLowerCase` is useful 🤔

Reply:
Yes that makes sense. toLocaleStrings seems like the better option for consistent results (correctness doesn't actually matter much, as long as the same hashing occurs on write and subsequent reads).